site-src/geps/gep-724.md

```diff
@@ -0,0 +1,698 @@
+  // For example, use the following to select an HTTPRoute:
+  //
+  // routes:
+  //   kind: HTTPRoute
+  //
+  // Otherwise, if an alternative API group is desired, specify the desired
+  // group:
+  //
+  // routes:
+  //   group: acme.io
+  //   kind: FooRoute
+  //
+  // Support: Core
+  //
+  // +optional
+  // +kubebuilder:default=gateway.networking.k8s.io
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Group *string `json:"group,omitempty"`
+  // Kind is the kind of the route resource to select.
+  //
+  // Kind MUST correspond to kinds of routes that are compatible with the
+  // application protocol specified in the Listener's Protocol field.
+  //
+  // If an implementation does not support or recognize this
+  // resource type, it SHOULD set the "ResolvedRefs" condition to false for
+  // this listener with the "InvalidRoutesRef" reason.
+  //
+  // Support: Core
+  Kind string `json:"kind"`
+```
+
+#### Added
+Note: The ListMapKey annotation for listeners would also have to change to name
+for this.
+
+```go
+type Listener struct {
+    // Name is the name of the Listener. If more than one Listener is present
+    // each Listener MUST specify a name. The names of Listeners MUST be unique
+    // within a Gateway.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    // +optional
+    Name *string `json:"name,omitempty"`
+    // ...
+}
+```
+
+The RouteBindingSelector struct would be renamed to ListenerRoutes, and a Kinds
+field would be added. Note that the Selector, Group, and Kind field would be
+removed from this struct as described above.
+
+```go
+type ListenerRoutes struct {
+  // ...
+  // Kinds specifies the groups and kinds of Routes that are allowed to bind to
+  // this Gateway listener. When unspecified or empty, the only limitation on
+  // the kinds of Routes supported is the Listener protocol. Kind MUST
+  // correspond to kinds of Routes that are compatible with the application
+  // protocol specified in the Listener's Protocol field. If an implementation
+  // does not support or recognize this resource type, it SHOULD set the
+  // "ResolvedRefs" condition to false for this listener with the
+  // "InvalidRoutesRef" reason.
+  //
+  // Support: Core
+  //
+  // +optional
+  // +kubebuilder:validation:MaxItems=10
+  Kinds []RouteGroupKind `json:"kinds,omitempty"`
+}
+
+type RouteGroupKind struct {
+  // Group is the group of the Route.
+  //
+  // Support: Core
+  //
+  // +optional
+  // +kubebuilder:default=gateway.networking.k8s.io
+  // +kubebuilder:validation:MaxLength=253
+  Group *string `json:"group,omitempty"`
+  // Kind is the kind of the Route.
+  //
+  // Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Kind string `json:"kind"`
+}
+```
+
+### Gateway Status
```

Comment:
@jpeach @youngnick @hbagdi I've added a new status section to describe suggested additions to status. I think this is relevant to different comments from all of you throughout the review of this GEP, so wanted to get your feedback on this addition if you have time.

Reply:
This is great, love it.